sycl/include/CL/sycl/detail/pi.h

```diff
@@ -440,7 +440,17 @@ typedef enum {
 
 typedef enum {
   // Device-specific value opaque in PI API.
-  PI_MEM_ADVISE_UNKNOWN
+  PI_MEM_ADVISE_UNKNOWN,
+  PI_MEM_ADVISE_CUDA_SET_READ_MOSTLY = 1,
+  PI_MEM_ADVISE_CUDA_UNSET_READ_MOSTLY = 2,
+  PI_MEM_ADVISE_CUDA_SET_PREFERRED_LOCATION = 3,
+  PI_MEM_ADVISE_CUDA_UNSET_PREFERRED_LOCATION = 4,
+  PI_MEM_ADVISE_CUDA_SET_ACCESSED_BY = 5,
+  PI_MEM_ADVISE_CUDA_UNSET_ACCESSED_BY = 6,
+  PI_MEM_ADVISE_CUDA_SET_PREFERRED_LOCATION_HOST = 7,
+  PI_MEM_ADVISE_CUDA_UNSET_PREFERRED_LOCATION_HOST = 8,
+  PI_MEM_ADVISE_CUDA_SET_ACCESSED_BY_HOST = 9,
+  PI_MEM_ADVISE_CUDA_UNSET_ACCESSED_BY_HOST = 10,
```

Comment:
These are all CUDA specific, however most other PI enums are built around the OpenCL counter-part. Granted there are no OpenCL descriptors yet, I would still prefer that the CUDA-specific options are moved to start at a number that isn't 1 as we should leave some buffer for OpenCL descriptors in the future.